### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

File: src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ static GapicInterfaceConfig createInterfaceConfig(
     for (CollectionConfigProto collectionConfigProto : interfaceConfigProto.getCollectionsList()) {
       String entityName = collectionConfigProto.getEntityName();
       ResourceNameConfig resourceName = resourceNameConfigs.get(entityName);
-      if (resourceName == null || !(resourceName instanceof SingleResourceNameConfig)) {
```
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java]: intellij said this was redundant

File: src/main/java/com/google/api/codegen/config/ProtoMethodModel.java [hunk 2]
```diff
@@ -208,11 +207,6 @@ public String getScopedDescription() {
     return outputFields;
   }
 
-  @Override
-  public List<ProtoField> getResourceNameInputFields() {
```
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/config/ProtoMethodModel.java]: this was never called

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/config/ProtoField.java [hunk 3]
```diff
@@ -41,7 +42,11 @@
   public ProtoField(Field protoField) {
     Preconditions.checkNotNull(protoField);
     this.protoField = protoField;
-    this.protoTypeRef = ProtoTypeRef.create(protoField.getType());
+    TypeRef typeRef = protoField.getType();
+    if (typeRef == null) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/config/ProtoField.java]: What case is this new condition handling? Could (should) protoField.getType() ever by null?

File: src/main/java/com/google/api/codegen/util/ProtoParser.java [hunk 4]
```diff
@@ -72,8 +73,19 @@ public OperationTypes getLongRunningOperation(Method method) {
...
+            .getOptionFields()
+            .keySet()
+            .stream()
+            .filter(o -> o.getFullName().equals("google.api.method_signature"))
+            .findAny();
+    if (!signatureDescriptor.isPresent()) {
+      return ImmutableList.of();
+    }
     MethodSignature methodSignature =
-        method.getDescriptor().getMethodAnnotation(AnnotationsProto.methodSignature);
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/util/ProtoParser.java]: Does this NPE, or return null? If it returns null, it seems simpler than using streams, and doesn't hard-code the "google.api.method_signature" string.

File: src/main/java/com/google/api/codegen/config/SingleResourceNameConfig.java [hunk 5]
```diff
@@ -83,7 +83,7 @@ public static SingleResourceNameConfig createSingleResourceName(
   // e.g. createPathTemplate("bookShelves/*/books/{book}") returns
   // PathTemplate.create("bookShelves/{bookShelf}/books/{book}")
   @VisibleForTesting
-  static String escapePathTemplate(String template) {
+  public static String escapePathTemplate(String template) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/config/SingleResourceNameConfig.java]: Why does this need to be public?

File: src/main/java/com/google/api/codegen/config/ProtoField.java (hunk 3 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/config/ProtoField.java]: Hmm I thought that this was necessary some time ago, but I just ran tests without this change and it's fine. Reverted.

File: src/main/java/com/google/api/codegen/util/ProtoParser.java (hunk 4 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/util/ProtoParser.java]: The problem with the previous implementation is that `MethodSignature methodSignature` would always return a non-null object, which makes it impossible to differentiate between an empty method signature (with 0 parameters) and the non-presence of a method signature. 

We need to do the stream thing to see if the method_signature annotation is even present.

Added a code comment to explain.

File: src/main/java/com/google/api/codegen/config/SingleResourceNameConfig.java (hunk 5 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/config/SingleResourceNameConfig.java]: Removed.

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/util/ProtoParser.java (hunk 4 above)
Comment (michaelbausor) [reply src/main/java/com/google/api/codegen/util/ProtoParser.java]: Hmm, it is a bit annoying that we can't access the underlying MethodDescriptorProto, as then we could call getOptions().hasExtension(...), similar to how the implementation of getMethodAnnotation calls getOptions().getExtension(...).

Does getOptionsFields().contains(AnnotationsProto.methodSignature.defaultInstance()) work?

Comment (andreamlin) [reply src/main/java/com/google/api/codegen/util/ProtoParser.java]: Oh that does work! Thanks! It's a lot cleaner.
Also added a test for empty method signatures.

**Comment by andreamlin** [comment]
PTAL

File: src/test/java/com/google/api/codegen/util/ProtoParserTest.java [hunk 6]
```diff
@@ -201,6 +201,24 @@ public void testGetMethodSignatures() {
         .containsExactly("name", "message", "string_builder");
   }
 
+  @Test
+  public void testEmptySignature() {
+    // Test that we can detect empty method signatures.
```
Comment (michaelbausor) [inline src/test/java/com/google/api/codegen/util/ProtoParserTest.java]: Can we also test the case when there is no method signature? (or is that already tested?)

Comment (andreamlin) [reply src/test/java/com/google/api/codegen/util/ProtoParserTest.java]: Sure! Done.
